pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -225,6 +225,11 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPRouteConfig(env *m
 func generateVirtualHostDomains(service *model.Service, port int, node *model.Proxy) []string {
 	domains := []string{string(service.Hostname), fmt.Sprintf("%s:%d", service.Hostname, port)}
 	domains = append(domains, generateAltVirtualHosts(string(service.Hostname), port, node.DNSDomain)...)
+	// TODO: can node.DNSDomain be integrated with the META?  Feels wrong to integrate the
+	// variables in that way...
+	for _, domain := range model.GetProxyMetadataDNSDomains(node) {
+		domains = append(domains, generateAltVirtualHosts(string(service.Hostname), port, domain)...)
```

Comment:
as stated above, convert node.DNSDomain to array and call this function for each domain in the list.
Also what happens if generateAltVirtualHosts returns nil ? do you end up adding nil to the domains array?

Reply:
not sure if generateAltVirtualHosts returns nil - will work on a UT for that.

Reply:
If it returns nil, append will not add any thing

Reply:
thanks - addressed.